compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -29,6 +29,64 @@ pub(super) fn opt_const_param_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<
         let parent_node = tcx.hir().get(parent_node_id);
 
         match parent_node {
+            // This matches on types who's paths couldn't be resolved without typeck'ing e.g.
```

Comment:
I was reading the [rustdoc for the `opt_const_param_of` query](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.opt_const_param_of) and I think this isn't *quite* right. 

This query would, I think, be invoked on the `3` in your example, and it would be returning the `const N1: usize` node.

It doesn't have much to do with `Self::Assoc` in particular, from what I can tell, except that this is the context in which the `3` appears.

Reply:
In other words, I think if you wrote `<Self as Foo>::Assoc<3>` the query could still be invoked

Reply:
OK, I'm wrong, I see your point. The *query* is invoked on `3`, but this **match arm** is specific to the case where the `3` appears as part of a `Self::Assoc<3>` type thing.

Reply:
(I am leaving my stream of consciousness to help guide you in how to improve the comment)

Reply:
I rewrote that comment to put more emphasis on what some of the variables correspond to and what exactly the match arm matches on :)